src/dune_rules/merlin.ml

```diff
@@ -120,7 +130,8 @@ module Processed = struct
     in
     if String.need_quoting s then Filename.quote s else s
 
-  let to_dot_merlin stdlib_dir pp_configs flags obj_dirs src_dirs extensions =
+  let to_dot_merlin stdlib_dir pp_configs flags obj_dirs src_dirs extensions
+      melc_flags =
```

Comment:
Do we need to pass these melange flags separately? They can be just a part of `flags`.

Reply:
It is not possible, or I don't see how. We talked about this earlier, please check https://github.com/ocaml/dune/pull/6476#discussion_r1026292933 for details.

Reply:
I pushed a commit describing what I meant.

Reply:
I see, thanks, I missed that `flags` in this function has some different semantics. Updated test in https://github.com/ocaml/dune/pull/6476/commits/437fabd9ae8254af1b62cdec23b6d091c90d6c94.